src/components/Navbar.js

```diff
@@ -21,29 +24,30 @@ export function Navbar() {
     }
   };
 
-  const handleTitleChange = (e) => {
-const handleTitleBlur = async () => {
-const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);
-debouncedSetLoading(true);
-  setIsEditingTitle(false);
-  if (editedTitle.trim() && editedTitle !== boardTitle) {
-    try {
-      await updateBoardTitle(editedTitle.trim());
-    } catch (error) {
-      setEditedTitle(boardTitle);
-      console.error('Failed to update board title:', error);
-      // Add user feedback mechanism here
-    }
-  } else {
-    setEditedTitle(boardTitle);
-  }
-};
+  const handleTitleBlur = async () => {
+    const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);

const handleTitleBlur = async () => {
```
The debounced loading state setter is created on every handleTitleBlur call. Move it outside the function to prevent recreation on each call.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;